diff -bruN --exclude .svn odtPHP0.10.orig/library/odf.php odtPHP0.10/library/odf.php
--- odtPHP0.10.orig/library/odf.php	2009-06-09 17:35:46.000000000 +0200
+++ odtPHP0.10/library/odf.php	2009-06-11 08:41:00.000000000 +0200
@@ -38,7 +38,7 @@
      * @param string $filename nom du fichier odt
      * @throws OdfException
      */
-    public function __construct($filename, $config = array())
+    public function __construct($filename, $config = array(), $tmpdir="/tmp")
     {
     	if (! is_array($config)) {
     		throw new OdfException('Configuration data must be passed as array');
@@ -53,6 +53,7 @@
         }
         $zipHandler = $this->config['ZIP_PROXY'];
         $this->file = new $zipHandler();
+        $this->file->TMP_DIR = $tmpdir;
         if ($this->file->open($filename) !== true) {
             throw new OdfException("Error while Opening the file '$filename' - Check your odt file");
         }
@@ -146,7 +147,7 @@
      *
      * @return void
      */
-    private function _parse()
+    protected function _parse()
     {
         $this->contentXml = str_replace(array_keys($this->vars), array_values($this->vars), $this->contentXml);
     }
diff -bruN --exclude .svn odtPHP0.10.orig/library/zip/PclZipProxy.php odtPHP0.10/library/zip/PclZipProxy.php
--- odtPHP0.10.orig/library/zip/PclZipProxy.php	2009-06-09 17:35:46.000000000 +0200
+++ odtPHP0.10/library/zip/PclZipProxy.php	2009-06-11 08:17:58.000000000 +0200
@@ -19,7 +19,7 @@
  */
 class PclZipProxy implements ZipInterface
 {
-	const TMP_DIR = './tmp';
+	public $TMP_DIR = './tmp';
 	protected $openned = false;
 	protected $filename;
 	protected $pclzip;
@@ -46,8 +46,8 @@
 		if (true === $this->openned) {
 			$this->close();
 		}		
-		if (!file_exists(self::TMP_DIR)) {
-			mkdir(self::TMP_DIR);
+		if (!file_exists($this->TMP_DIR)) {
+			mkdir($this->TMP_DIR);
 		}
 		$this->filename = $filename;
 		$this->pclzip = new PclZip($this->filename);
@@ -90,11 +90,11 @@
 		}
 		$localname = preg_replace("/(?:\.|\/)*(.*)/", "\\1", $localname);
 		$localpath = dirname($localname);
-		$tmpfilename = self::TMP_DIR . '/' . basename($localname);
+		$tmpfilename = $this->TMP_DIR . '/' . basename($localname);
 		if (false !== file_put_contents($tmpfilename, $contents)) {
 			$this->pclzip->delete(PCLZIP_OPT_BY_NAME, $localname);
 			$add = $this->pclzip->add($tmpfilename,
-				PCLZIP_OPT_REMOVE_PATH, self::TMP_DIR,
+				PCLZIP_OPT_REMOVE_PATH, $this->TMP_DIR,
 				PCLZIP_OPT_ADD_PATH, $localpath);
 			unlink($tmpfilename);
 			if (!empty($add)) {
@@ -121,17 +121,17 @@
 		if (isSet($localname)) {
 			$localname = preg_replace("/(?:\.|\/)*(.*)/", "\\1", $localname);
 			$localpath = dirname($localname);
-			$tmpfilename = self::TMP_DIR . '/' . basename($localname);
+			$tmpfilename = $this->TMP_DIR . '/' . basename($localname);
 		} else {
 			$localname = basename($filename);
-			$tmpfilename = self::TMP_DIR . '/' . $localname;
+			$tmpfilename = $this->TMP_DIR . '/' . $localname;
 			$localpath = '';
 		}
 		if (file_exists($filename)) {
 			copy($filename, $tmpfilename);
 			$this->pclzip->delete(PCLZIP_OPT_BY_NAME, $localname);
 			$this->pclzip->add($tmpfilename,
-				PCLZIP_OPT_REMOVE_PATH, self::TMP_DIR,
+				PCLZIP_OPT_REMOVE_PATH, $this->TMP_DIR,
 				PCLZIP_OPT_ADD_PATH, $localpath);
 			unlink($tmpfilename);
 			return true;
@@ -149,9 +149,9 @@
 		}		
 		$this->pclzip = $this->filename = null;
 		$this->openned = false;
-		if (file_exists(self::TMP_DIR)) {
-			$this->_rrmdir(self::TMP_DIR);
-			rmdir(self::TMP_DIR);
+		if (file_exists($this->TMP_DIR)) {
+			$this->_rrmdir($this->TMP_DIR);
+			rmdir($this->TMP_DIR);
 		}
 		return true;
 	}
